reflex/state.py

```diff
@@ -923,8 +979,12 @@ def __getattribute__(self, name: str) -> Any:
             **super().__getattribute__("inherited_vars"),
             **super().__getattribute__("inherited_backend_vars"),
         }
-        if name in inherited_vars:
-            return getattr(super().__getattribute__("parent_state"), name)
+
+        # For now, handle router_data updates as a special case.
```

Comment:
Is this because router_data is only attached to the top level state?

Reply:
Before we used to push `router_data` into all substates, but that was a kludge that i introduced a long time ago while working on cached_var. Ultimately we only want to keep one copy of `router_data`, and we keep that on the base `rx.State` now.

The reason why `router_data` needs this special case is because it's present in `get_skip_vars`, so it doesn't get the normal "inherited var" tracking because it's treated as a non-var. Contrast this with `router`, which is always considered an inherited var and handled in the regular way.

The long term fix to remove this special case is to separate "internal" vars from "non-serialized" vars. I think that might actually happen sooner than later when the "Remove Cloudpickle" PR comes in and refactors how serialization is handled.